test/ZNSSubRegistrar.test.ts

```diff
@@ -88,14 +88,21 @@ describe("ZNSSubRegistrar", () => {
         zeroVaultAddress: zeroVault.address,
       });
       // Give funds to users
+      const users = [
+        rootOwner,
+        lvl2SubOwner,
+      ];
+      const totalAdminBalance = await zns.zToken.balanceOf(admin.address);
+      const userBalanceInitial = totalAdminBalance / BigInt(users.length);
+      await zns.zToken.connect(admin).approve(await zns.treasury.getAddress(), totalAdminBalance);
```

Comment:
Only the person calling to register a domain needs to approve the treasury, like the registrant below is `rootOwner` and they are calling to approve on L105

Reply:
I agree. does admin here actually needs to approve treasury? is he registering any domains in the test?